068-persistent-bottom-nav-bar/lib/ui/views/main/main_view.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:flutter/material.dart';
+import 'package:persistent_bottom_nav_bar/app/app.router.dart';
+import 'package:persistent_bottom_nav_bar/ui/views/main/main_viewmodel.dart';
+import 'package:stacked/stacked.dart';
+
+class MainView extends StatelessWidget {
+  const MainView({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return ViewModelBuilder<MainViewModel>.reactive(
+      builder: (context, model, child) => WillPopScope(
+        onWillPop: () async {
+          // If it's not on the first route in the bottom nav bar
+          if (model.currentIndex != 0) {
+            // Then pop to the first index
+            model.setIndex(0);
+
+            // And return false to prevent the default behavior of a back button
+            return false;
+          }
+          // Let the default behavior handle the back button tap
+          return true;
+        },
+        child: Scaffold(
+          body: Stack(
+            children: [
+              _MainBody(
```

Comment:
there's a better way to do this by using an internal navigator per view stack. This is not a good solution. It's not something I'd want us to teach. 

Could you research how to use generated routers for all nested view paths and see if that provides the same result. We can make the PostsView have its own Navigator and the same for the UsersView